VideoApp/VideoApp/NewCode/Views/FocusLayout/FocusLayoutViewModel.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import TwilioVideo
+import Combine
+
+class FocusLayoutViewModel: ObservableObject {
+    struct Presenter {
+        var identity: String = ""
+        var presentationTrack: VideoTrack?
+    }
+
+    @Published var isPresenting = false
+    @Published var presenter = Presenter()
+    @Published var dominantSpeaker = ParticipantViewModel()
+    private var roomManager: RoomManager!
+    private var subscriptions = Set<AnyCancellable>()
+
+    func configure(roomManager: RoomManager) {
+        self.roomManager = roomManager
+        
+        roomManager.roomConnectPublisher
+            .sink { [weak self] in self?.update() }
+            .store(in: &subscriptions)
+        
+        roomManager.roomDisconnectPublisher
+            .sink { [weak self] _ in self?.update() }
+            .store(in: &subscriptions)
+
+        roomManager.remoteParticipantConnectPublisher
+            .sink { [weak self] _ in self?.update() }
+            .store(in: &subscriptions)
+
+        roomManager.remoteParticipantDisconnectPublisher
+            .sink { [weak self] _ in self?.update() }
+            .store(in: &subscriptions)
+
+        roomManager.remoteParticipantChangePublisher
+            .sink { [weak self] _ in self?.update() }
+            .store(in: &subscriptions)
+    }
+
+    private func update() {
+        guard let presenter = findPresenter() else {
+            isPresenting = false
+            presenter = Presenter()
+            dominantSpeaker = ParticipantViewModel()
+            return
+        }
+
+        isPresenting = true
+        self.presenter = presenter
+        dominantSpeaker = findDominantSpeaker()
+        dominantSpeaker.isDominantSpeaker = false // No need to distinguish in the UI since only 1 participant is shown
+    }
+    
+    private func findPresenter() -> Presenter? {
+        guard let presenter = roomManager.remoteParticipants.first(where: { $0.presentationTrack != nil }) else {
+            return nil
+        }
+        
+        return Presenter(identity: presenter.identity, presentationTrack: presenter.presentationTrack!)
```

Comment:
@ceaglest for this video layout that currently displays the dominant speaker and presentation track I think I might need to be setting track priority. I could set the track priority of the presentation track to high. The dominant speaker track priority is currently set to server default. So will both tracks end up being high priority which kind of defeats the purpose of priority? Maybe only the presentation track should be high priority for a layout like this.

Reply:
Will talk about this one when he is back in the office.